cumulusci/tasks/salesforce/users.py

```diff
@@ -0,0 +1,135 @@
+import base64
+import os
+import json
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import CumulusCIException
+
+
+class UploadDefaultUserProfilePhoto(BaseSalesforceApiTask):
+    task_docs = """
+
+    """
+
+    task_options = {
+        "photo_path": {
+            "description": "Path to desired profile photo.",
+            "required": True,
+        }
+    }
+
+    def get_user_id(self):
+        user_id = self.sf.restful("")["identity"][-18:]
+        self.logger.info(
+            f"Uploading profile photo for the default User with ID {user_id}."
+        )
+        return user_id
+
+    def _run_task(self):
+        # Get the User Id of the targeted user.
+        user_id = self.get_user_id()
+
+        # Upload profile photo ContentDocument.
+        path = self.options["photo_path"]
+
+        self.logger.info(f"Setting user photo to {path}")
+        with open(path, "rb") as version_data:
```

Comment:
I'd recommend turning the string path into a pathlib.Path instance: `path = pathlib.Path(self.options["photo_path"])`.

Check `path.exists()` to confirm the file exists and show a helpful error if not.

And then pathlib.Path has some useful attributes and methods to use below:
- `path.name` instead of `os.path.split(path)[-1]`
- `path.stem` instead of `os.path.splitext(os.path.split(path)[-1])[0]`
- `path.read_bytes()` to get the file contents in a single call without needing to use `open` as a context manager.

Reply:
Will do.

Reply:
pushed changes to use`pathlib`.   This is much simpler!  ✅